frontend/web/components/navigation/TabMenu/Tabs.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useCallback, useMemo, useRef, useState } from 'react'
+import ModalHR from 'components/modals/ModalHR'
+import Utils from 'common/utils/utils'
+import { useHistory } from 'react-router-dom'
+import TabButton from './TabButton'
+
+import classNames from 'classnames'
+import DropdownMenu from 'components/base/DropdownMenu'
+import { useOverflowVisibleCount } from 'common/hooks/useOverflowVisibleCount'
+
+interface TabItemProps {
+  tabLabel: React.ReactNode
+  tabLabelString?: string
+  'data-test'?: string
+  id?: string
+  className?: string
+  children: React.ReactNode
+}
+
+interface TabsProps {
+  children:
+    | React.ReactElement<TabItemProps>
+    | React.ReactElement<TabItemProps>[]
+  onChange?: (index: number) => void
+  theme?: 'tab' | 'pill'
+  uncontrolled?: boolean
+  value?: number
+  className?: string
+  urlParam?: string
+  hideNavOnSingleTab?: boolean
+  buttonTheme?: string
+  noFocus?: boolean
+  isRoles?: boolean
+  history?: any
+  overflowX?: boolean
+}
+
+const MORE_BTN_SIZE = 34
+const EXTRA_RESERVE = MORE_BTN_SIZE + 20
```

Comment:
Why not just:

```suggestion
const EXTRA_RESERVE = 54
```

Also, I appreciate I have no idea what I'm doing when it comes to UI, but I do always find it a little unnerving when we have hard-coded values like this. Could you explain a little about what we're doing here?